lib/core.js

```diff
@@ -1780,6 +1780,25 @@ interface $Iterator<+Yield,+Return,-Next> extends $Iterable<Yield,Return,Next> {
      * @param callbackfn A function that accepts up to two arguments: the yielded value, and it's index. The map method calls the callbackfn function once per iteration. The callback must return an iterable.
      */
     flatMap<U>(callbackfn: (value: Yield, index: number) => $Iterable<U, mixed, void>): $Iterator<U, void, mixed>;
+    /**
+     * Calls the argument "reducer" callback function on each value produced by this iterator, passing in the return value from the calculation on the preceding iteration. The final result of running the reducer across all elements is a single value.
+     * @param callbackfn A function that accepts up to three arguments: the accumulated value, the current value, and the current index.
+     * @param initialValue If initialValue is specified, it is used as the initial value to start
+     * the accumulation. If omitted, the first call of the callback gets the first two iterated values.
+     */
+    reduce<U>(
```

Comment:
I believe this is a more accurate type for `reduce` without an initial value than  [the type we have for `Array.prototype.reduce`](https://github.com/facebook/flow/blob/efe35182d024b61d1c5a897eba77e9df23844f45/lib/core.js#L863) and the similar [type that TypeScript uses for `Iterator.prototype.reduce`](https://github.com/microsoft/TypeScript/blob/d85767abfd83880cea17cea70f9913e9c4496dcc/src/lib/esnext.iterator.d.ts#L69).

See motivating example in test. It likely rarely matters in practice. Happy to switch to the simpler type if I missed something, or if we just prefer simplicity.